core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java

```diff
@@ -297,4 +372,53 @@ private void checkNamespaceValidity(String namespace, String errorMsg) throws Se
             throw new ServerException(errorMsg);
         }
     }
+
+    private boolean systemRamLimitExceeded() {
+        String ramUsage = null;
+        try {
+            ramUsage = dockerConnector.getSystemInfo().ramUsage();
+        } catch (IOException e) {
+            LOG.error("A problem occurred while getting system information from docker", e);
+        }
+        if (ramUsage == null) {
+            return false;
+        }
+        String[] ramValues = ramUsage.split("/ ");
+        if (ramValues.length < 2) {
+            LOG.error("A problem occurred while parsing system information from docker");
+            return false;
+        }
+        long ramUsed = getBytesAmountFromString(ramValues[0]);
+        long ramTotal = getBytesAmountFromString(ramValues[1]);
+        return ((ramUsed * 100) / ramTotal > 90);
+    }
+
+    private void sendMessage(String line) {
+        final ChannelBroadcastMessage bm = new ChannelBroadcastMessage();
+        bm.setChannel("resources_chanel");
+        bm.setBody(JsonUtils.getJsonString(line));
+        try {
+            WSConnectionContext.sendMessage(bm);
+        } catch (Exception e) {
+            LOG.error("A problem occurred while sending websocket message", e);
+        }
+    }
+
+    private long getBytesAmountFromString(String string) {
+        if (string.contains("KiB")) {
```

Comment:
Can you give an example of the output?
It seems that you can reuse: [this class](https://github.com/eclipse/che/blob/867ec652424cd23a587f164f6acf3e8e44158bdf/core/commons/che-core-commons-lang/src/main/java/org/eclipse/che/commons/lang/Size.java#L22-L22) for parsing